### PR TITLE
WAITP-1148: Project Landing Page Loader

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Fix the state where projects haven't loaded and the single project template flashes before properly rendering the multi project template

### CODE REVIEW

File: packages/web-frontend/src/screens/LandingPage/LandingPageFooter.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { Link, Typography, List } from '@material-ui/core';
 import { useCustomLandingPages } from './useCustomLandingPages';
 
 const Footer = styled.footer`
+  margin-top: auto;
```
Comment (tcaiger) [inline packages/web-frontend/src/screens/LandingPage/LandingPageFooter.js:12]: this keeps the footer at the bottom in the flex column before the content of the landing pages has loaded

File: packages/web-frontend/src/screens/LandingPage/LandingPage.js [hunk 2]
```diff
@@ -53,7 +54,12 @@ export const LandingPage = () => {
       <TopBar />
       <Wrapper backgroundImage={backgroundImage}>
         <Container maxWidth={false}>
-          {projects.length > 1 ? <MultiProjectLandingPage /> : <SingleProjectLandingPage />}
+          {/* tupaia requires projects to work so we can assume that if there are no projects, it's just */}
+          {/* because they haven't loaded yet. We can replace this with more idiomatic loading state */}
+          {/* when we refactor to use react-query */}
+          {projects.length === 0 && <LoadingScreen isLoading background={null} />}
```
Comment (tcaiger) [inline packages/web-frontend/src/screens/LandingPage/LandingPage.js:60]: I'm not sure if there is a better way to do this 3 way logic but I didn't want to use a nested ternary so I did this

File: packages/web-frontend/src/screens/LoadingScreen/index.js [hunk 3]
```diff
@@ -8,21 +8,35 @@
...
+  top: 0;
+  left: 0;
+  height: 100vh;
+  width: 100vw;
+  display: flex;
+  background: ${props => props.$background};
+  align-items: center;
+  justify-content: center;
+  z-index: ${Number.MAX_SAFE_INTEGER};
+`;
```
Comment (acdunham) [inline packages/web-frontend/src/screens/LoadingScreen/index.js:26]: Yay for styled-components!

**Review by acdunham — APPROVED** [review]
Looks good to me. Agree that a 3 way ternary is hard to read, and we can work on a good loading state later on

**Pull request merged**